ansible_collections/arista/avd/roles/eos_designs/python_modules/ip_addressing/__init__.py

```diff
@@ -0,0 +1,346 @@
+
+from collections import ChainMap
+from functools import cached_property
+import ipaddress
+import importlib
+from ansible_collections.arista.avd.plugins.plugin_utils.avdfacts import AvdFacts
+from ansible_collections.arista.avd.plugins.plugin_utils.utils import AristaAvdError, get
+
+
+DEFAULT_AVD_IP_ADDRESSING_PYTHON_MODULE = "ansible_collections.arista.avd.roles.eos_designs.python_modules.ip_addressing"
+DEFAULT_AVD_IP_ADDRESSING_PYTHON_CLASS_NAME = "AvdIpAddressing"
+
+
+class AvdIpAddressing(AvdFacts):
+
+    '''
+    Class used to render IP addresses either from custom Jinja2 templates or using default Python Logic
+
+    Since some templates might contain certain legacy variables (switch_*),
+    those are mapped from the switch.* model
+
+    This class is imported adhoc based on the variable `templates.ip_addressing.python_module` so it can
+    be overridden by a custom python class.
+    '''
+
+    def _ip(self, pool: str, prefixlen: int, subnet_offset: int, ip_offset: int) -> str:
+        network = ipaddress.ip_network(pool, strict=False)
+        subnets = network.subnets(new_prefix=prefixlen)
+        if isinstance(subnets, list):
```

Comment:
This sounds weird! :) the doc from 3.6 document iterators - do you have some traceback on this?

Reply:
I don't. But I *think* it was because of this: https://github.com/python/cpython/issues/71870
I was running some 3.8 version at the time.